core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -118,10 +118,17 @@ private static Executor defaultBlockingTaskExecutor() {
     }
 
     private static final class DefaultBlockingTaskExecutorHolder {
-        static final Executor INSTANCE = new ThreadPoolExecutor(
-                0, DEFAULT_MAX_BLOCKING_TASK_THREADS,
-                60, TimeUnit.SECONDS, new LinkedTransferQueue<>(),
-                new DefaultThreadFactory("armeria-blocking-tasks", true));
+        static final Executor INSTANCE;
+
+        static {
+            final ThreadPoolExecutor instance = new ThreadPoolExecutor(
```

Comment:
Ah - found the queue is the difference :) Didn't realize newCachedThreadPool doesn't queue up tasks when threads are full that seems like a weird default.